core/new-gui/src/app/workspace/service/workflow-graph/model/workflow-action.service.ts

```diff
@@ -391,6 +459,9 @@ export class WorkflowActionService {
         }
       },
     };
+
+    const commandMessage: CommandMessage = { action: "deleteOperator", parameters: [operatorID], type: "execute" };
+    this.sendCommand(JSON.stringify(commandMessage));
     this.executeAndStoreCommand(command);
```

Comment:
I see for every endpoint you have a `command`, the original logic is to `executeAndStoreComand(command)`. now since we need to sync this `command` to backend, so that other frontend can be synced, you are also sending a `commandMessage` to the backend. So there are two steps now, 1) send `commandMessage` and 2) execute and store the `command` locally.

I wonder can we make the `sendCommand` call inside `executeAndStoreCommand` since it can be treated as part of execution life cycle. basically merge the two steps into one.

Also It is not very clear now, if one of them fails what should be done. for example, if the commandMessage is not being sent successfully due to some network issue, what will the behavior be?

Reply:
I thought of the same thing when I was migrating the code, but some actions has `sendCommand`, but do not include `executeAndStoreCommand`. Maybe leave the refactoring till later.
As for the scenario you're suggesting, the frontend changes will not be propagated in this case, and to come up with its solutions we might need further discussion.

Reply:
agreed this might need further discussion, seems not trivial to me. we can resolve this for now.

Reply:
I added a TODO for this.